File: ruby/word_count.rb
class Phrase
  def initialize(phrase)
    @phrase=sanitize(phrase)
    @words={}
  end

  def word_count
    @phrase.each do |word|
      @words[word]=@phrase.count(word)
    end
    @words
  end

  def sanitize(phrase)
    phrase=phrase.downcase.gsub(/[\W&&[^']]/, " ").split(" ").each do |word|
      word.gsub!(/[']/, "") if word.start_with?("'") || word.end_with?("'")
    end
  end
end

File: js/word-count.js
Words = function(){};

Words.prototype.count = function(phrase) {
  words = sanitize(phrase)
  return wordCount(words);
};

function sanitize(phrase){
  return phrase.toLowerCase().replace(/(?!('t|'s))\W/g, " ").split(/\s+/).filter(x=>x!=="");
};

function wordCount(words){
  set = {};
  words.forEach(function(word){
    set[word]===undefined ? set[word]=1 : set[word]+=1
  })
  return set
};

module.exports = Words;
